auth/principal.go

```diff
@@ -145,3 +145,68 @@ type User interface {
 
 	setRolesSince(ch.TimedSet)
 }
+
+// PrincipalUpdates represents a set of changes to a principal. It is meant for use with db.UpdatePrincipal, but is in
+// the auth package to avoid import cycles.
+// Name must always be specified, the other fields can be omitted to keep their current value.
+type PrincipalUpdates struct {
+	Name             string
+	ExplicitChannels *base.Set
```

Comment:
base.Set is already a map, so already supports nil values, and we should be able to differentiate between undefined and empty set without defining as a pointer to base.Set.